app/src/lib/get-main-guid.ts

```diff
@@ -0,0 +1,45 @@
+import { app } from 'electron'
+import { readFile, writeFile } from 'fs/promises'
+import { join } from 'path'
+import { uuid } from './uuid'
+
+let cachedGUID: string | null = null
+
+/** Get the GUID for the Main process. */
+export async function getMainGUID(): Promise<string> {
+  if (!cachedGUID) {
+    cachedGUID = await readGUIDFile()
+  }
+
+  return cachedGUID
+}
+
+/** Reads the persisted GUID from the .guid file or generates a new one */
+export async function readGUIDFile(): Promise<string> {
+  let guid = undefined
+
+  try {
+    guid = await readFile(getGUIDPath(), 'utf8')
```

Comment:
Can we trim the read contents first? Just so that a newline introduced by curious folks opening the file and saving it doesn't invalidate it?